components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AMDefaultKeyManagerImpl.java

```diff
@@ -424,7 +426,13 @@ public OAuthApplicationInfo retrieveApplication(String consumerKey) throws APIMa
         }
 
         try {
-            ClientInfo clientInfo = dcrClient.getApplication(consumerKey);
+            encoded = false;
+            String isConsumerKeyEncoded = System.getProperty(ENCODE_CONSUMER_KEY, "false");
+            if (isConsumerKeyEncoded.equalsIgnoreCase("true")) {
```

Comment:
Shall we swap this to avoid NPE?

Reply:
The default value is set to be "false" in the above method to get the system property, hence it can not become null

Reply:
But better to have it the other way around as a best practice. Shall we change it?

Reply:
+1, or we have to first convert it to boolean and check with `== true`.

Reply:
Addressed in b1a9d7e66f2d8ee05ee815b7e2d632dc91eeb734